### PR TITLE
Handle scoped modules from extraNodeModules config

### PR DESCRIPTION
**Pull request by sfrdmn**
**Summary**

Metro allows configuring `extraNodeModules` for additional module resolution mappings. It chokes on scoped module names (e.g. `@org/module`), because it doesn't compute the path correctly

**Test plan**

```
// In metro.config.js
extraNodeModules: {
  '@org/module': path.resolve(__dirname, "wherever")
}

// In consuming code
import module from '@org/module'
```

Assuming the module is neither in `node_modules` nor comes from the Haste map, the former code will fail without this patch

### CODE REVIEW

File: packages/metro-resolver/src/resolve.js [hunk 1]
```diff
@@ -103,8 +103,15 @@ function resolve(
   const extraPaths = [];
   const {extraNodeModules} = context;
   if (extraNodeModules) {
-    const bits = path.normalize(moduleName).split(path.sep);
-    const packageName = bits[0];
+    let bits = path.normalize(moduleName).split(path.sep);
+    let packageName;
+    // Normalize packageName and bits for scoped modules
+    if (bits[0] && bits[0][0] === '@') {
```
Comment (rafeca) [inline packages/metro-resolver/src/resolve.js]: Can you do `if (bits.length >= 2 && bits[0].startsWith('@')) {` here?

**Comment by rafeca** [comment]
This is awesome! Thanks for the fix!

Can you please add a simple test for this? We currently only have integration tests for the resolution logic, but it should be easy to add a new test using the following one as a base: https://github.com/facebook/metro/blob/master/packages/metro/src/DeltaBundler/__tests__/traverseDependencies-integration-test.js#L1506 ?

**Comment by sfrdmn** [comment]
Updated, but tests failing on master it seems?

**Comment by rafeca** [comment]
Thanks! Looks good!

Yup, we have an unrelated issue with the tests on node v6

**Pull request closed without merging**